### PR TITLE
feat(storage): add list method

### PR DESCRIPTION
**Pull request by rsivan**
Add 'list(options)' method to AngularFireStorageReference, to further reflect the firebase storage api.

Fixes: #2801

### Checklist

   - Issue number for this PR: #2801
   - Docs included?: no 
   - Test units included?: no 
   - In a clean directory, `yarn install`, `yarn test` run successfully? yes

### Description

Expose list method as part of AngularFireStorageReference.

### Code sample
This is well documented in the firebase storage documentation.

### CODE REVIEW

File: src/storage/ref.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import { ListResult, Reference, SettableMetadata, StringFormat, UploadMetadata } from './interfaces';
+import { ListResult, Reference, SettableMetadata, StringFormat, UploadMetadata, ListOptions } from './interfaces';
```
Comment (yasinkocak) [inline src/storage/ref.ts:1]:
```suggestion
import { ListOptions, ListResult, Reference, SettableMetadata, StringFormat, UploadMetadata } from './interfaces';
```
Alphabetized

**Comment by rsivan** [comment]
Is there anything I can do to advance this pull request?

**Comment by jamesdaniels** [comment]
I'll look into this for the next minor release. Sorry was blocked by working on 6.1.5 and 7.0.0-alpha.

**Comment by jamesdaniels** [comment]
I'll look into this for the next minor release. Sorry was blocked by working on 6.1.5 and 7.0.0-alpha.

**Comment by jamesdaniels** [comment]
Closing this in favor of the rebased #2960

**Comment by jamesdaniels** [comment]
Thanks for the contribution

**Pull request closed without merging**